stub/src/main/java/io/grpc/stub/RpcMethod.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.stub;
+
+import io.grpc.MethodDescriptor.MethodType;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotates a method descriptor method to provide metadata for annotation processing.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.METHOD)
+public @interface RpcMethod {
```

Comment:
I'm not wild about the "RpcMethod" name, since we don't have anything "Rpc"-prefixed already. ... but I also can't think of anything better. @carl-mastrangelo, @zhangkun83, @dapengzhang0, @ericgribkoff, @zpencer, any thoughts on another name? We could throw it in a annotations package, but we'd still want the name to avoid colliding.

Reply:
+1 for annotations package, then I am fine with `RpcMethod`.

Reply:
Moved to stub.annotations package.

Reply:
I guess I'm fine with the name. If we want another one on the service it would be RpcService. The only main competitor would probably be "GrpcMethod" and "GrpcService", although I would also sort of expect that to be more likely to collide with others (users of gRPC that make their own annotations for injection-based systems). So I think the "Rpc" is good.